### PR TITLE
test(python): Mark some tests as slow

### PR DESCRIPTION
**Pull request by stinodego**
I noticed some tests running very slowly. I profiled the tests using `pytest --durations 20` and marked all tests slower than 0.1 seconds as `slow`.

I also eliminated the use of `from_repr` in unrelated tests. There's a lot of magic going on in that implementation and I don't want other tests to rely on it.

### CODE REVIEW

File: py-polars/tests/unit/dataframe/test_from_dict.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        "g": time(0, 1, 2),
+        "h": date(2023, 3, 14),
+        "i": timedelta(seconds=3601),
+        "j": datetime(2111, 11, 11, 11, 11, 11, 11),
+        "k": "「趣味でヒーローをやっている者だ」",
+    }
+    # note: deliberately set this value large; if all dtypes are
+    # on the fast-path it'll only take ~0.03secs. if it becomes
+    # even remotely noticeable that will indicate a regression.
+    # TODO: This is now slow (~0.15 seconds). Needs to be looked into.
```
Comment (stinodego) [inline py-polars/tests/unit/dataframe/test_from_dict.py:166]: Added a TO DO here. Apparently there is a regression here?

Comment (alexander-beedie) [reply py-polars/tests/unit/dataframe/test_from_dict.py:166]: I'll check that - that timing may only relate to release builds, as well as being M1-specific (I'm compiling something up now to see if there's a real/meaningful drift there or not)

Comment (alexander-beedie) [reply py-polars/tests/unit/dataframe/test_from_dict.py:166]: @stinodego: Good news, can confirm no regression here; it's just quite sensitive to being release/debug or not. A fresh release build shows 0.0260 secs for mw, whereas debug was ~0.14 secs - so it's all good.

Comment (stinodego) [reply py-polars/tests/unit/dataframe/test_from_dict.py:166]: All right, will remove the TO DO when I am refactoring some tests later! Thanks for looking into this 👍

Comment (alexander-beedie) [reply py-polars/tests/unit/dataframe/test_from_dict.py:166]: No problem; I recognised it as something I looked at before ;)

**Pull request merged**